torch/_dynamo/convert_frame.py

```diff
@@ -499,6 +497,7 @@ def log_bytecode(prefix, name, filename, line_no, code):
         if hooks.guard_export_fn is not None:
             hooks.guard_export_fn(output.guards)
 
+        output.local_scope.clear()
```

Comment:
How come?

Reply:
release memory, see `test_release_input_memory`

Reply:
I guess I have a lifecycle question, which is why isn't OutputGraph automatically going dead after compilation? If it went dead, it shouldn't be necessary to clear things off of it. If it stays live, it seems bad to store "heavy" state on it because then we have to clear it like this.

Reply:
As in, this PR seems like a "make things more readable" improvement, as opposed to a "I MUST have this to implement something later." And if it's the former, it doesn't seem worth it if you're gooping up the lifecycle story.

Reply:
> As in, this PR seems like a "make things more readable" improvement, as opposed to a "I MUST have this to implement something later." And if it's the former, it doesn't seem worth it if you're gooping up the lifecycle story.

It's the latter. I need access to f_locals/f_globals much earlier (OutputGraph) than where they are sent down today (CheckFnManager)

Reply:
> I guess I have a lifecycle question, which is why isn't OutputGraph automatically going dead after compilation?

This is a prior issue. OutputGraph already has a cleanup function, but we can't use it because it runs a little earlier - we need f_locals/f_globals here to be cleaned up at the end of the frame, as it is used in CheckFNManager.

```
    def cleanup(self) -> None:
        # There is a reference cycle between tracer and OutputGraph, causing
        # some of the tensor objects to be held alive for longer than necessary.
```